### ISSUE TITLE
Add support for \kern

### ISSUE THREAD

**Issue by Pomax** [Khan/KaTeX#422]
I'm using the following block of LaTeX, to make sure the maths makes visual sense to the reader, but KaTeX does not expect `\hskip` in this block, and errors out:
```
\begin{align*}
  linear &= \hskip{2.5em} 1 + 1 \\
  square &= \hskip{1.7em} 1 + 2 + 1\\
  cubic &= \hskip{0.85em} 1 + 3 + 3 + 1\\
  hypercubic &= 1 + 4 + 6 + 4 + 1
\end{align*
```
The error it generates is: 
```
Error: KaTeX parse error: Expected & or \\ or \end at position 42: gned}linear &= ̲\hskip{2.5em} 
```

**Comment by kevinbarabash** [comment]
In this situation you could also use `\kern` which we also don't support yet, but is easier to implement because we don't have to worry about line splitting or stretch/squash of glue.  I'm going to rename the issue to `implement \kern` with the expectation that you should be able to use `\kern` anywhere.

**Comment by Pomax** [comment]
had to look that one up (yay, TeX by Topic), but yeah that would be fine too.

**Comment by WChargin** [comment]
@Pomax A few unsolicited comments about your original TeX!

First, it's either `\hskip2.5em` or `\hspace{2.5em}`; the former is a TeX primitive while the latter is a LaTeX command. Second, you'll certainly want to enclose the labels on the left in `\mathrm`s so that the spacing is correct—e.g., on the `li` pair in "linear."

Third, and most importantly, doing horizontal spacing manually like this is probably a Bad Idea™. When I change the font to mathpazo, everything will break. When you're in `\scriptsize`, everything will break. The proper way to do this is to think about the layout you want: a left-aligned column with the names, then a column with an equals sign, then a centered column with the values. This calls for an `array`. So you might try this:
```tex
\begin{equation*}
  \begin{array}{lcc}
    \mathrm{linear}     &=& 1 + 1  \\
    \mathrm{square}     &=& 1 + 2 + 1 \\
    \mathrm{cubic}      &=& 1 + 3 + 3 + 1 \\
    \mathrm{hypercubic} &=& 1 + 4 + 6 + 4 + 1
  \end{array}
\end{equation*}
```
This will almost work, but the spacing around the equals signs will be way too big, because it will be getting the normal column gutter spacing instead of the proper spacing for a relation:
![First attempt: spacing around equals signs is too large](https://cloud.githubusercontent.com/assets/4317806/14673434/ed27b4fe-06cb-11e6-945c-8f1cb7244ee4.png)

Instead, then, we should tell TeX not to add spacing around the center column by changing its column specification to `@{}c@{}` (which indicates "at left: nothing; at right: nothing"), and then add empty atoms to the left and right of the equals sign so that it's treated as a relation:
```tex
\begin{equation*}
  \begin{array}{l@{}c@{}c}
    \mathrm{linear}     &{}={}& 1 + 1  \\
    \mathrm{square}     &{}={}& 1 + 2 + 1 \\
    \mathrm{cubic}      &{}={}& 1 + 3 + 3 + 1 \\
    \mathrm{hypercubic} &{}={}& 1 + 4 + 6 + 4 + 1
  \end{array}
\end{equation*}
```
This works quite well:
![Second attempt: spacing cleaned up](https://cloud.githubusercontent.com/assets/4317806/14673498/3f368aa4-06cc-11e6-91da-6bab35da9c2f.png)

Finally, we can make the body code a bit nicer by pulling the `\mathrm`s and empty atoms into the column specifications:
```tex
\begin{equation*}
  \begin{array}{>{\mathrm\bgroup}l<{\egroup}@{}>{{}}c<{{}}@{}c}
    linear     &=& 1 + 1  \\
    square     &=& 1 + 2 + 1 \\
    cubic      &=& 1 + 3 + 3 + 1 \\
    hypercubic &=& 1 + 4 + 6 + 4 + 1
  \end{array}
\end{equation*}
```
If this is new to you, you should parse the column specifications as

  * `>{\mathrm\bgroup}l<{\egroup}`: a `l`eft-aligned column, to whose content we should prepend `\mathrm{` and to whose content we should append `}`;
  * `@{}`: empty gutter (and no space) between these two columns
  * `>{{}}c<{{}}`: a `c`entered column, to whose content we should prepend and append an empty atom `{}`;
  * `@{}`: empty gutter; and
  * a centered column.

This results in the same output.

KaTeX supports only the first variant of these (:cry:) because advanced column specs don't seem to be supported. But if you're using this in a normal (La)TeX document then the last one is the way to go.

**Comment by Pomax** [comment]
Hey @WChargin, thanks for that!

Although as you point out, I'm mostly concerned about "what works online" right now. For offline document generation there are way too many commands that don't work in either MathJax or KaTeX, so I'm mostly looking at this from the perspective of having 50 pages of web content with formulae that currently use MathJax, that I'd like to switch over to KaTeX. 

If the `equasion*` and `array` environments are properly supported by KaTeX, then it's probably worth updating the code on my site over, as it's by far cleaner, but the added typesetting precision of the last form feels like overkill for web presentation. Good to know about for offline document generation, but the lack of typesetting precision in the text around the latex blocks makes getting things absolutely perfect in the latex itself just not worth it =)

**Comment by gagern** [comment]
Implemented as of #423.
